### PR TITLE
Task6

### PR DESCRIPTION
**Pull request by aleksandr-zharov**
Task: https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task6-sqs-sns/task.md
FE: https://dndbrxqx66t48.cloudfront.net/
Evaluation criteria:
 1 - File serverless.yml contains configuration for catalogBatchProcess function
 2 - File serverless.yml contains policies to allow lambda catalogBatchProcess function to interact with SNS and SQS
 3 - File serverless.yml contains configuration for SQS catalogItemsQueue
 4 - File serverless.yml contains configuration for SNS Topic createProductTopic and email subscription
Additional (optional) tasks
 +1 - catalogBatchProcess lambda is covered by unit tests
 +1 - set a Filter Policy for SNS createProductTopic in serverless.yml (Create an additional email subscription and distribute messages to different emails depending on the filter for any product attribute)

### CODE REVIEW

File: import-service/handlers/catalog-batch-process.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import {SQSEvent} from 'aws-lambda';
+import {SNS} from 'aws-sdk';
+import {errorHandler, successHandler} from "../common/default";
+import {createProduct} from "../../product-service/services/products";
+
+export const handler = async (event: SQSEvent) => {
```
Comment (DzmitryU) [inline import-service/handlers/catalog-batch-process.ts:6]: This handler must be implemented in **products-service**

Comment (aleksandr-zharov) [reply import-service/handlers/catalog-batch-process.ts]: Yes, you are right. Thanks. Fixed + updated PR description.

File: product-service/handlers/catalog-batch-process.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      if (!validationResult.error) {
+        const result = await createProduct(product);
+        list.push(result);
+      } else {
+        console.error('Invalid product', product);
+      }
+    }
+
+    const sns = new SNS();
+    sns.publish({
```
Comment (DzmitryU) [inline product-service/handlers/catalog-batch-process.ts:24]: You could use await.publish(...).promise() to get rid of callbacks

File: import-service/handlers/import-file-parser.ts [hunk 3]
```diff
@@ -23,7 +23,19 @@ export const handler = async (event: S3Event) => {
         s3Stream
             .pipe(csv())
             .on('data', (data) => {
-              console.log(JSON.stringify(data, null, 4));
+              console.log('importFileParser::parse:', JSON.stringify(data, null, 4));
```
Comment (DzmitryU) [inline import-service/handlers/import-file-parser.ts:26]: You could make utils promisify pipeline + queue.sendMessage(...).promise() to get rid of callbacks

**Pull request merged**